Content.Server/Database/ServerDbBase.cs

```diff
@@ -505,16 +505,23 @@ public async Task<ServerBanExemptFlags> GetBanExemption(NetUserId userId)
         public async Task EditServerRoleBan(int id, string reason, NoteSeverity severity, DateTimeOffset? expiration, Guid editedBy, DateTimeOffset editedAt)
         {
             await using var db = await GetDb();
+            var banTime = await db.DbContext.RoleBan
+                .Where(b => b.Id == id)
+                .Select(b => b.BanTime)
+                .SingleOrDefaultAsync();
 
-            var ban = await db.DbContext.RoleBan.SingleOrDefaultAsync(b => b.Id == id);
-            if (ban is null)
+            if (banTime == default)
                 return;
-            ban.Severity = severity;
-            ban.Reason = reason;
-            ban.ExpirationTime = expiration?.UtcDateTime;
-            ban.LastEditedById = editedBy;
-            ban.LastEditedAt = editedAt.UtcDateTime;
-            await db.DbContext.SaveChangesAsync();
+
+            await db.DbContext.RoleBan
+                .Where(b => b.BanTime == banTime)
+                .ExecuteUpdateAsync(setters => setters
+                    .SetProperty(b => b.Severity, severity)
+                    .SetProperty(b => b.Reason, reason)
+                    .SetProperty(b => b.ExpirationTime, expiration.HasValue ? expiration.Value.UtcDateTime : (DateTime?)null)
+                    .SetProperty(b => b.LastEditedById, editedBy)
+                    .SetProperty(b => b.LastEditedAt, editedAt.UtcDateTime)
+                );
         }
```

Comment:
Initially I did have it as a loop doing an update on each of the bans it queried from a list that matched the same initial  ban time. But my thought behind this was it was more a compound SQL query, where one query is done vs the 3-10 SQL queries it defines on the loop and update for.

Although I also have no idea how it does the translation to the SQL under the hood works, it may be the same amount of updates but this seemed like it read a little cleaner.

Either way performance wise It didn't feel like it was going to be a big impactor on average its only going to update about 3-10 rows on its use and its a vary rarely used action to modify a roleban.

Reply:
Let me know if its a blocker.